### DISCUSSION TITLE
templ fmt and whitespace for readability

### DISCUSSION THREAD

**Discussion by pharrisee** [a-h/templ#884] [General]
Would there be any way of keeping inserted whitespace (blank lines), when templ code is formatted within VSCode?

As an example, given this code:

```
if sg.Icon == "" {
    <span></span>
} else {
    <img src={ fmt.Sprintf("/static/icon/%s.svg", sg.Icon) } class="w-6 h-6"/>
}

<div>
    { children... }
</div>
```

The blank line in the middle is there simply to aid readability of the code.  It would be nice for that to stay in the formatted code.

`templ generate` should still behave as it currently does, this is just cosmetic.

**Comment by a-h** [comment]
There's an issue for this at https://github.com/a-h/templ/issues/374 - just haven't implemented it due to higher priority things to do. Will take a PR if someone gets around to it though.
